packages/flutter/test/material/expansion_panel_test.dart

```diff
@@ -109,4 +110,90 @@ void main() {
+    const Duration kSizeAnimationDuration = const Duration(milliseconds: 1000);
+    Widget build(bool a, bool b, bool c) {
+      return new MaterialApp(
+        home: new Column(
+          children: <Widget>[
+            new ExpansionPanelList(
+              animationDuration: kSizeAnimationDuration,
+              children: <ExpansionPanel>[
+                new ExpansionPanel(
+                  headerBuilder: (BuildContext context, bool isExpanded) => const Placeholder(),
+                  body: const SizedBox(height: 100.0, child: const Placeholder()),
+                  isExpanded: a,
+                ),
+                new ExpansionPanel(
+                  headerBuilder: (BuildContext context, bool isExpanded) => const Placeholder(),
+                  body: const SizedBox(height: 100.0, child: const Placeholder()),
+                  isExpanded: b,
+                ),
+                new ExpansionPanel(
+                  headerBuilder: (BuildContext context, bool isExpanded) => const Placeholder(),
+                  body: const SizedBox(height: 100.0, child: const Placeholder()),
+                  isExpanded: c,
+                ),
+              ],
+            ),
+          ],
+        ),
+      );
+    }
+
+    await tester.pumpWidget(build(false, false, false));
+    expect(tester.renderObjectList(find.byType(AnimatedSize)), hasLength(3));
+    expect(tester.getRect(find.byType(AnimatedSize).at(0)), new Rect.fromLTWH(0.0, 56.0, 800.0, 0.0));
+    expect(tester.getRect(find.byType(AnimatedSize).at(1)), new Rect.fromLTWH(0.0, 113.0, 800.0, 0.0));
+    expect(tester.getRect(find.byType(AnimatedSize).at(2)), new Rect.fromLTWH(0.0, 170.0, 800.0, 0.0));
+
+    await tester.pump(const Duration(milliseconds: 200));
+    expect(tester.getRect(find.byType(AnimatedSize).at(0)), new Rect.fromLTWH(0.0, 56.0, 800.0, 0.0));
+    expect(tester.getRect(find.byType(AnimatedSize).at(1)), new Rect.fromLTWH(0.0, 113.0, 800.0, 0.0));
+    expect(tester.getRect(find.byType(AnimatedSize).at(2)), new Rect.fromLTWH(0.0, 170.0, 800.0, 0.0));
+
+    await tester.pumpWidget(build(false, true, false));
+    expect(tester.getRect(find.byType(AnimatedSize).at(0)), new Rect.fromLTWH(0.0, 56.0, 800.0, 0.0));
+    expect(tester.getRect(find.byType(AnimatedSize).at(1)), new Rect.fromLTWH(0.0, 113.0, 800.0, 0.0));
+    expect(tester.getRect(find.byType(AnimatedSize).at(2)), new Rect.fromLTWH(0.0, 170.0, 800.0, 0.0));
+
+    await tester.pump(kSizeAnimationDuration ~/ 2);
+    expect(tester.getRect(find.byType(AnimatedSize).at(0)), new Rect.fromLTWH(0.0, 56.0, 800.0, 0.0));
+    // The MaterialGaps animate in using kThemeAnimationDuration (hardcoded),
+    // which should be less than our test size animation length. So we can assume that they
+    // appear immediately. Here we just verify that our assumption is true.
+    expect(kThemeAnimationDuration, lessThan(kSizeAnimationDuration ~/ 2));
```

Comment:
nit: I'd move this check closer to the definition of `kSizeAnimationDuration` as it ensures a static invariant that does not depend on the state the test is in.

Reply:
fixed